core/src/main/scala/quasar/destination/gbq/GBQFlow.scala

```diff
@@ -281,18 +343,15 @@ final class GBQFlow[F[_]: Concurrent](
               }
             }
 
-            uploadStream.pull.echo
-
+            uploadStream.pull.echo >> 
+            checkJobStatus(job, config.timeOut, config.maxRetries, 0).pull.echo
```

Comment:
I think this statement would make more sense in the main `ingest` method, rather than here in `upload`